fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -214,8 +214,14 @@ public void setMaintainHistory(boolean maintainHistory) {
 
     @Nonnull
     protected RecordMetaData buildMetaData(RecordMetaDataProto.MetaData metaDataProto, boolean validate) {
+        return buildMetaData(metaDataProto, validate, null);
+    }
+
+    @Nonnull
+    protected RecordMetaData buildMetaData(RecordMetaDataProto.MetaData metaDataProto, boolean validate, @Nullable Descriptors.FileDescriptor evolvedDescriptor) {
```

Comment:
Discussion copied from #276:

@alecgrieser:
> I'm going back and forth in my head whether the evolved descriptor should be a member of the `FDBMetaDataStore` or a parameter like this.
> 
> It kind of seems to me is that what we need is either an `FDBMetaDataStore` provider or builder. (We might want a builder anyway. If you'll notice, there is not a way of specifying both a `KeySpacePath` and a (non-null) `MetaDataCache` to this class.) I think you would want to be able to specify the `evolvedDescriptor` and maybe the seed meta-data as well. I think that would be necessary to re-use the same object across multiple `FDBRecordStore` instances. (This is analogous to the problems that we ran into with the `KeySpacePath`.)
> 
> @MMcM, do you have thoughts on this?

@amouehsan:

> Yeah I noticed that, while working on the cache. I like the idea of making it a member. It already has dependencies / setDependencies too.

Reply:
It looks like this became a member variable eventually, which sounds like a defensible decision to me.